gulpfile.js

```diff
@@ -11,112 +11,146 @@
-var del = require('del');
-var eslint = require('gulp-eslint');
-
-var path = require('path');
-
-var minimalDocument = require('./util/minimalDocument');
-
-var micro = "polymer-micro.html";
-var mini = "polymer-mini.html";
-var max = "polymer.html";
-var workdir = 'dist';
-
-var distMicro = path.join(workdir, micro);
-var distMini = path.join(workdir, mini);
-var distMax = path.join(workdir, max);
-
-var pkg = require('./package.json');
-
-var cleanupPipe = lazypipe()
-  // remove leading whitespace and comments
-  .pipe(polyclean.leftAlignJs)
-  // remove html wrapper
-  .pipe(minimalDocument)
-  // Add real version number
-  .pipe(replace, /(Polymer.version = )'master'/, '$1"' + pkg.version + '"')
-;
-
-function vulcanizeWithExcludes(target, excludes) {
-  if (excludes) {
-    excludes = excludes.map(function(ex) { return path.resolve(ex); });
-  }
-  return function() {
-    return gulp.src(target)
-      .pipe(vulcanize({
-        stripComments: true,
-        excludes: excludes
-      }))
-      .pipe(cleanupPipe())
-      .pipe(gulp.dest(workdir));
-  };
-}
-
-gulp.task('micro', vulcanizeWithExcludes(micro));
-gulp.task('mini', vulcanizeWithExcludes(mini, [micro]));
-gulp.task('max', vulcanizeWithExcludes(max, [mini, micro]));
+/* global require */
+const gulp = require('gulp');
+const gulpif = require('gulp-if');
+const audit = require('gulp-audit');
+const rename = require('gulp-rename');
+const runseq = require('run-sequence');
+const del = require('del');
+const eslint = require('gulp-eslint');
+const fs = require('fs');
+const path = require('path');
+
+const DIST_DIR = 'dist';
+const BUNDLED_DIR = path.join(DIST_DIR, 'bundled');
+const UNBUNDLED_DIR = path.join(DIST_DIR, 'unbundled');
+const DEFAULT_BUILD_DIR = BUNDLED_DIR;
+const POLYMER_LEGACY = 'polymer.html';
+const POLYMER_ELEMENT = 'polymer-element.html';
+const DEFAULT_BUILD_TARGET = POLYMER_LEGACY;
+const ENTRY_POINTS = [POLYMER_LEGACY, POLYMER_ELEMENT];
+
+const polymer = require('polymer-build');
+const PolymerProject = polymer.PolymerProject;
+const project = new PolymerProject({ shell: DEFAULT_BUILD_TARGET });
+const fork = polymer.forkStream;
+
+const mergeStream = require('merge-stream');
+const babel = require('gulp-babel');
+const uglify = require('gulp-uglify');
+const htmlmin = require('gulp-htmlmin');
+const gzipSize = require('gzip-size');
+const prettyBytes = require('pretty-bytes');
 
 gulp.task('clean', function() {
-  return del(workdir);
+  return del(DIST_DIR);
+});
+
+gulp.task('build', ['clean'], () => {
+ // process source files in the project
+ const sources = project.sources()
+   .pipe(project.splitHtml())
+   // add compilers or optimizers here!
+   .pipe(gulpif(/\.js$/, babel({presets: ['es2015']/*, plugins: ['external-helpers']*/})))
+   .pipe(gulpif(/\.js$/, uglify()))
+   .pipe(project.rejoinHtml())
+   .pipe(htmlmin({removeComments: true}));
+
+ // process dependencies
+ const dependencies = project.dependencies()
+   .pipe(project.splitHtml())
```

Comment:
Since sources and dependencies are processed the same, maybe extract a function to improve maintainability?